src/main/java/com/google/devtools/build/lib/analysis/ConfiguredTargetFactory.java

```diff
@@ -304,6 +308,76 @@ private ConfiguredTarget createRule(
                     prerequisiteMap.values()))
             .build();
 
+    if (!"toolchain".equals(rule.getRuleClass())) {
+      boolean isIncompatible = false;
+
+      // This is incompatible if one of the dependencies is as well.
+      for (ConfiguredTargetAndData infoCollection : prerequisiteMap.values()) {
+        if (infoCollection.getConfiguredTarget().getProvider(IncompatiblePlatformProvider.class) != null) {
+          isIncompatible = true;
+        }
+      }
+
+      // This is incompatible if explicitly specified to be.
+      if (!isIncompatible && ruleContext.attributes().has("target_compatible_with")) {
+        Iterable<ConstraintValueInfo> constraintValues =
+            PlatformProviderUtils.constraintValues(
+                ruleContext.getPrerequisites("target_compatible_with", TransitionMode.DONT_CHECK));
+        for (ConstraintValueInfo constraintValue : constraintValues) {
+          if (!ruleContext.targetPlatformHasConstraint(constraintValue)) {
```

Comment:
I can't remember, but what's the intended outcome if a rule is `target_compatible_with = ["x86", "ppc"]`? That it's only compatible with target platforms that have both CPUs, or target platforms with either of those CPUs?

Reply:
I thought we split that discussion out into a separate doc?  I thought the proposal was compatible with all of the platform constraints.  Which wouldn't make much sense here, since nothing can match both.  And the proposed solution for handling OR logic was to use select.

Reply:
What @AustinSchuh said sounds right to me. The current design as I understand it assumes that all constraints have to be satisfied. From the doc:
```python
cc_library(
    name = "mylib",
    target_compatible_with = select({
        ":is_linux": ["@platforms//cpu:arm"],
        ":is_windows": ["@platforms//cpu:x86_64"],
  })
```
Perhaps a "pure" OR would be kind of an identity?
```python
cc_library(
    name = "mylib",
    target_compatible_with = select({
        ":is_linux": ["@platforms//os:linux"],
        ":is_windows": ["@platforms//os:windows"],
  })
```
I realize that I haven't thought about it too much, but it's definitely worth adding a test for this and documenting how to do OR operations.

Reply:
I added a test for OR logic and it seems to work well for that one simple example.
It was the first time using `constraint_value` targets in `select()` statements. Luckily I found #8583 which told me what to do :)
I think for now that should be sufficient for a "version 1.0".